xarray/tests/test_backends.py

```diff
@@ -3744,6 +3749,25 @@ def test_chunk_key_encoding_v2(self) -> None:
                 assert actual["var1"].encoding["chunks"] == (2, 2)
 
 
+@requires_zarr_v3
+class TestZarrNoConsolidatedMetadataSupport(ZarrBase):
+    @contextlib.contextmanager
+    def create_zarr_target(self):
+        class NoConsolidatedMetadataSupportStore(WrapperStore[Store]):
```

Comment:
Defined in here because I didn't want to deal with trying to make this class definable if v3 isn't installed